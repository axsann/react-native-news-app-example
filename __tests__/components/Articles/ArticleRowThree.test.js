// @flow

import 'react-native'
import React from 'react'
import ArticleRowThree from '../../../src/components/Articles/ArticleRowThree.component'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

const mockThreeItems = [
  {"title":"あああああああああああああああああああああああああああ","url":"https://example.com","date":"20XX年Y月Z日","imgUrl":"https://placehold.jp/150x150.png","tags":["タグB"]},
  {"title":"あああああああああああああああああああああああああああ","url":"https://example.com","date":"20XX年Y月Z日","imgUrl":"https://placehold.jp/150x150.png","tags":["タグA","タグG"]},
  {"title":"あああああああああああああああああああああああああああ","url":"https://example.com","date":"20XX年Y月Z日","imgUrl":"https://placehold.jp/150x150.png","tags":["タグE"]},
]

describe('ArticleRowThree', () => {
  it('renders correctly', () => {
    const instance = renderer.create(
      <ArticleRowThree articles={mockThreeItems} onPressArticle={()=>{}} />
    )

    expect(instance.toJSON()).toMatchSnapshot()
  })
})
