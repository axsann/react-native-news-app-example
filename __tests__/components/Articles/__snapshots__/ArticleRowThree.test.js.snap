// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleRowThree renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flexDirection": "row",
      "height": 190,
      "marginTop": 10,
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flex": 1,
        "marginLeft": 10,
        "marginRight": 10,
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "https://placehold.jp/150x150.png",
          }
        }
        style={
          Object {
            "flex": 8,
          }
        }
      />
      <View
        style={
          Array [
            Object {
              "flex": 9,
            },
            Object {
              "marginTop": 10,
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#000000",
              "fontSize": 12.5,
              "fontWeight": "bold",
            }
          }
        >
          あああああああああああああああああああああああああああ
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 10,
              "marginTop": 5,
            }
          }
        >
          20XX年Y月Z日
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#eeeeee",
        "width": 1,
      }
    }
  />
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flex": 1,
        "marginLeft": 10,
        "marginRight": 10,
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "https://placehold.jp/150x150.png",
          }
        }
        style={
          Object {
            "flex": 8,
          }
        }
      />
      <View
        style={
          Array [
            Object {
              "flex": 9,
            },
            Object {
              "marginTop": 10,
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#000000",
              "fontSize": 12.5,
              "fontWeight": "bold",
            }
          }
        >
          あああああああああああああああああああああああああああ
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 10,
              "marginTop": 5,
            }
          }
        >
          20XX年Y月Z日
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#eeeeee",
        "width": 1,
      }
    }
  />
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flex": 1,
        "marginLeft": 10,
        "marginRight": 10,
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "https://placehold.jp/150x150.png",
          }
        }
        style={
          Object {
            "flex": 8,
          }
        }
      />
      <View
        style={
          Array [
            Object {
              "flex": 9,
            },
            Object {
              "marginTop": 10,
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#000000",
              "fontSize": 12.5,
              "fontWeight": "bold",
            }
          }
        >
          あああああああああああああああああああああああああああ
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 10,
              "marginTop": 5,
            }
          }
        >
          20XX年Y月Z日
        </Text>
      </View>
    </View>
  </View>
</View>
`;
